adapter/internal/oasparser/model/resource.go

```diff
@@ -111,3 +116,66 @@ func CreateMinimalDummyResourceForTests(path string, methods []Operation, id str
 		sandboxUrls:    sandboxUrls,
 	}
 }
+
+// Custom sort implementation to sort the Resources based on the resource path
+type byPath []Resource
+
+//Len Returns the length of the arry
+func (a byPath) Len() int { return len(a) }
+
+//Less  returns true if the first item is less than the second parameter
+func (a byPath) Less(i, j int) bool {
+	// Get the less weighted path.
+	// Paths can be in several types.
+	// - /pet
+	// - /pet/{id}
+	// - /pet/index.html
+	// - /pet/{id}/price
+	// - /pet/*
+	// When representing these resources in envoy configuration, they must be ordered correctly.
+	// Sorted order
+	// - /pet/index.html
+	// - /pet
+	// - /pet/{id}
+	// - /pet/{id}/price
+	// - /pet/{id}/{price}
+	// - pet/*
+	// Considerations...
+	// The concrete paths are matched first
+	// Any path with . character gets higher precidence
+	// Precedence decreases when the number of path parameters increses.
+	// The wild card path is matched last.
+
+	// Replace all the non symbol characters with empty string ("") Because the alphabetical order is not mandatory.
+	charMatcher := regexp.MustCompile(`[\w\s]`)
+	pathI := charMatcher.ReplaceAllString(a[i].path, "")
+	pathJ := charMatcher.ReplaceAllString(a[j].path, "")
+
+	// if wildcard is matched for either i or j, it will be returned as greater.
+	wildCardMatcher := regexp.MustCompile(`(\/[*]$)`)
+	if wildCardMatcher.Match([]byte(pathI)) || wildCardMatcher.Match([]byte(pathJ)) {
+		return !wildCardMatcher.Match([]byte(pathI)) || wildCardMatcher.Match([]byte(pathJ))
+	}
+
+	// if the dot is matched (either i or j), the path is considered less than the other one.
+	// If both i and j match this at the same time, compare the full path.
+	dotMatcher := regexp.MustCompile(`\.`)
```

Comment:
Shall we consider `.{}` pattern and `{}.`. I think that will simplify the logic because the path param with the dot is the concern.

Reply:
Here is the output

```
/pet
/pet/{id}
/pet/index.html
/pet/{id}/price
/pet/{id}/{price}
/pet/*
/pet/{petId}.com
/pet/pet.{anc}
/pet/{pet}.{anc}
```
Sorted>>>>>>>>>>>>>>>>>


```
/pet/index.html
/pet/pet.{anc}
/pet/{petId}.com
/pet/{pet}.{anc}
/pet
/pet/{id}
/pet/{id}/price
/pet/{id}/{price}
/pet/*
```